### PR TITLE
Add alarm integration

### PR DESCRIPTION
**Pull request by iMicknl**

### CODE REVIEW

File: custom_components/tahoma/alarm_control_panel.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        await self.async_execute_command(COMMAND_SET_ALARM_STATUS, STATE_DETECTED)
+
+    async def async_alarm_arm_custom_bypass(self, code=None) -> None:
+        """Send arm custom bypass command."""
+        await self.async_execute_command(COMMAND_SET_ALARM_STATUS, STATE_UNDETECTED)
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        # if self.device.widget == "TSKAlarmController":
```
Comment (iMicknl) [inline custom_components/tahoma/alarm_control_panel.py:161]: Not sure if this is the best way, but I could not figure out yet which state shows that someone is actually using the controller...

File: custom_components/tahoma/alarm_control_panel.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+        if self.has_state(MYFOX_ALARM_STATUS_STATE):
+            state = self.select_state(MYFOX_ALARM_STATUS_STATE)
+
+            if state == STATE_ARMED:
+                alarm_state = STATE_ALARM_ARMED_AWAY
+            elif state == STATE_DISARMED:
+                alarm_state = STATE_ALARM_DISARMED
+            elif state == STATE_PARTIAL:
+                alarm_state = STATE_ALARM_ARMED_NIGHT
```
Comment (tetienne) [inline custom_components/tahoma/alarm_control_panel.py:83]: You can use a dict (constant) to map the states from Somfy to HA.

File: custom_components/tahoma/alarm_control_panel.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            state = self.select_state(INTERNAL_CURRENT_ALARM_MODE_STATE)
+
+            if state == STATE_OFF:
+                alarm_state = STATE_ALARM_DISARMED
+            elif state == STATE_ZONE_1:
+                alarm_state = STATE_ALARM_ARMED_HOME
+            elif state == STATE_ZONE_2:
+                alarm_state = STATE_ALARM_ARMED_NIGHT
+            elif state == STATE_TOTAL:
+                alarm_state = STATE_ALARM_ARMED_AWAY
```
Comment (tetienne) [inline custom_components/tahoma/alarm_control_panel.py:95]: Same remark than above.

File: custom_components/tahoma/alarm_control_panel.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        if self.has_command(COMMAND_SET_ALARM_STATUS):
+            supported_features |= SUPPORT_ALARM_TRIGGER
+            supported_features |= SUPPORT_ALARM_ARM_CUSTOM_BYPASS
+
+        return supported_features
+
+    async def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        await self.async_execute_command(
+            self.select_command(COMMAND_DISARM, COMMAND_ALARM_OFF)
```
Comment (tetienne) [inline custom_components/tahoma/alarm_control_panel.py:140]: Why use select command, when you know the command?

File: custom_components/tahoma/alarm_control_panel.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        )
+
+    async def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        await self.async_execute_command(self.select_command(COMMAND_ALARM_PARTIAL_1))
```
Comment (tetienne) [inline custom_components/tahoma/alarm_control_panel.py:145]: Same remark.

File: custom_components/tahoma/alarm_control_panel.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        """Send arm custom bypass command."""
+        await self.async_execute_command(COMMAND_SET_ALARM_STATUS, STATE_UNDETECTED)
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        if self.device.widget == "TSKAlarmController":
+            return False
+
+        return True
```
Comment (tetienne) [inline custom_components/tahoma/alarm_control_panel.py:173]:
```suggestion
       return self.device.widget != "TSKAlarmController"
```

File: custom_components/tahoma/alarm_control_panel.py (hunk 4 above)
Comment (iMicknl) [reply custom_components/tahoma/alarm_control_panel.py:140]: I am not sure where I copied this from.. Thanks for spotting this.

Comment (iMicknl) [reply custom_components/tahoma/alarm_control_panel.py]: @tetienne you need it to select one of them, cover does the same.

**Review by vlebourl — APPROVED** [review]

**Review by tetienne — APPROVED** [review]

**Pull request merged**